File: sorting/bubble_sort/bubble_sort.cpp
#include <vector>

using namespace std; 


vector<int> sort(vector<int>& arr) {
    for(int i=0; i<arr.size(); i++){
        for(int j=0; j<arr.size(); j++){
            if(arr[j] > arr[j+1]){
                int temp = arr[j]; 
                arr[j] = arr[j+1];
                arr[j+1] = temp;
            }
        }
    }

    return arr; 
}

File: sorting/bubble_sort/bubble_sort.rs
fn sort(arr: &mut [i16]){
    let mut counter = 0; 
    
    'max_iter: loop{
        for index in 0..(arr.len()-1){
            if arr[index] > arr[index+1]{
                let temp: i16 = arr[index]; 
                arr[index] = arr[index+1]; 
                arr[index+1] = temp; 
            }
        }
    if counter == arr.len() - 1{
        break 'max_iter; 
        }
    else {counter += 1;}    
    }
}
